execution/binary/table.go

```diff
@@ -12,9 +13,37 @@ import (
 	"github.com/thanos-community/promql-engine/execution/parse"
 )
 
-type sample struct {
-	t int64
-	v float64
+type binOpSide string
+
+const (
+	lhBinOpSide binOpSide = "left"
+	rhBinOpSide binOpSide = "right"
+)
+
+type errManyToManyMatch struct {
+	sampleID          uint64
+	duplicateSampleID uint64
+	side              binOpSide
+}
+
+func (err errManyToManyMatch) Error() string {
```

Comment:
Are we using this anywhere?

Reply:
Good point. Initially this struct was implementing the error interface, but I don't think it's needed in the current version. Removed.